crates/hir_def/src/type_ref.rs

```diff
@@ -248,6 +332,110 @@ impl TypeRef {
             }
         }
     }
+
+    pub fn fold(&self, f: &mut impl FnMut(&TypeRef) -> Option<TypeRef>) -> TypeRef {
```

Comment:
We can either leave this here or remove it since I don't think its required for the expansion in `TypeRef -> Ty` lowering change.